test/taskcluster_secrets_api_test.js

```diff
@@ -0,0 +1,127 @@
+      args:       testValueExpires,
+      statusCode: 401 // It's not authorized!
+    },
+    {
+      testName:   "Captain (write only), fail to read.",
+      clientName: "captain-write",
+      apiCall:    "get",
+      name:        "captain:" + FOO_KEY,
+      statusCode: 401 // it's not authorized!
+    },
+    {
+      testName:   "Captain (read only), read foo.",
+      clientName: "captain-read",
+      apiCall:    "get",
+      name:       "captain:" + FOO_KEY,
+      res:        testValueExpires
+    },
+    {
+      testName:   "Captain, update allowed key",
+      clientName: "captain-write",
+      apiCall:    "update",
+      name:        "captain:" + FOO_KEY,
+      args:       testValueExpires2,
+      res:        {}
+    },
+    {
+      testName:   "Captain (read only), read updated foo.",
+      clientName: "captain-read",
+      apiCall:    "get",
+      name:        "captain:" + FOO_KEY,
+      res:        testValueExpires2
+    },
+    {
+      testName:   "Captain (write only), delete foo.",
+      clientName: "captain-write",
+      apiCall:    "remove",
+      name:        "captain:" + FOO_KEY,
+      res:        {}
+    },
+    {
+      testName:   "Captain (read only), read deleted foo.",
+      clientName: "captain-read",
+      apiCall:    "get",
+      name:        "captain:" + FOO_KEY,
+      statusCode: 404
+    },
+    {
+      testName:   "Captain (write only), delete already deleted foo.",
+      clientName: "captain-write",
+      apiCall:    "remove",
+      name:        "captain:" + FOO_KEY,
+      statusCode: 404
+    },
+    {
+      testName:   "Captain (write only), write bar that is expired.",
+      clientName: "captain-write",
+      apiCall:    "set",
+      name:        "captain:" + BAR_KEY,
+      args:       testValueExpired,
+      res:        {}
+    },
+    {
+      testName:   "Captain (read only), read bar that is expired.",
+      clientName: "captain-read",
+      apiCall:    "get",
+      name:        "captain:" + BAR_KEY,
+      statusCode: 410
+    },
+    {
+      testName:   "Captain (write only), delete bar.",
+      clientName: "captain-write",
+      apiCall:    "remove",
+      name:        "captain:" + BAR_KEY,
+      res:        {}
+    },
+  ]
+
+  for (let options of testData) {
+    test(options.testName, async function() {
+      let client = helper.clients[options.clientName];
+      let res = undefined;
+      try {
+        if (options.args) {
+          res = await client[options.apiCall](options.name, options.args);
+        } else {
+          res = await client[options.apiCall](options.name);
+        }
+      } catch (e) {
+        if (e.statusCode) {
+          assert.deepEqual(options.statusCode, e.statusCode);
+        } else {
+          throw e; // if there's no statusCode this isn't an API error
+        }
+      }
+      for (let key in options.res) {
+        assert.deepEqual(res[key], options.res[key]);
+      }
+    });
+  }
```

Comment:
I need two test cases:
 1) show that "set" is idempotent
 2) show that "set" doesn't naively overwrite existing items...ie. returns 409 if secret already exists and isn't a match.